application/src/main/java/org/togetherjava/tjbot/features/help/HelpSystemHelper.java

```diff
@@ -355,4 +357,36 @@ public String getTagName() {
             return tagName;
         }
     }
+
+    /**
+     * Retrieves a {@link HelpThreadsRecord} object from the database based on the specified channel
+     * ID.
+     * <p/>
+     * This method queries for a record in the HELP_THREADS table where the CHANNEL_ID column
+     * matches the provided channelID param. It returns an Optional containing the matching record.
+     *
+     * @param channelID is channel unique identifier
+     *
+     * @return An Optional containing the HelpThreadsRecord for the specified channel ID, or an
+     *         empty Optional if no matching record is found.
+     *
+     * @throws DataAccessException if something is wrong with database
+     *
+     */
+    Optional<HelpThreadsRecord> getThreadRecordByChannelId(final long channelID) {
```

Comment:
u should not return a `...Record` class, as that keeps the database connection alive. that class must not be persisted by anyone or used for longer than necessary. just create the actual helper method u need, namely `getOpByHelpThreadId` instead. Then u also dont need to query the DB for all data of the row, but only the data u actually need (author)

Reply:
> Then u also dont need to query the DB for all data of the row, but only the data u actually need (author)

Sorry I don't get this, would you like to give more explanation please 🤔

Reply:
`HelpThreadsRecord` is a class that must not be kept alive for longer than a few milliseconds. it holds the connection to the database open, as it has methods like `record.update()` or `record.delete()` which then feed back straight into the database.